compliance/collection/main.go

```diff
@@ -288,24 +302,8 @@ func main() {
 
 	// TODO(ROX-13935): Remove FakeNodeInventory and its FF
 	if features.RHCOSNodeScanning.Enabled() {
-		var analyzer nodeinventorizer.NodeInventorizer
-		if features.UseFakeNodeInventory.Enabled() {
-			log.Infof("Using FakeNodeInventorizer")
-			analyzer = &nodeinventorizer.FakeNodeInventorizer{}
-		} else {
-			log.Infof("Using NodeInventoryCollector")
-			analyzer = &nodeinventorizer.NodeAnalyzer{}
-		}
-
 		i := intervals.NewNodeScanIntervalFromEnv()
-		scanner := nodeinventorizer.NewCachingScanner(
-			analyzer,
-			"/cache/inventory-cache",
-			env.NodeScanningCacheDuration.DurationSetting(),
-			env.NodeScanningInitialBackoff.DurationSetting(),
-			env.NodeScanningMaxBackoff.DurationSetting(),
-			func(duration time.Duration) { time.Sleep(duration) })
-		nodeInventoriesC := manageNodeScanLoop(ctx, i, scanner)
+		nodeInventoriesC := manageNodeScanLoop(ctx, i, nodeInventoryClient)
```

Comment:
Now the issue is here. You guarded the assignment of `nodeInventoryClient`. It will be passed here as `nil` if it wasn't set before. 

I think you need to guard this call (and the goroutine below) if `nodeInventoryClient` is not `nil`.

```go
if nodeInventoryClient != nil {
		nodeInventoriesC := manageNodeScanLoop(ctx, i, nodeInventoryClient)
		go func() { /* ... */ }
}
```

Otherwise the panic will still happen since a `nil` `nodeInventoryClient` will be passed to `manageNodeScanLoop`. 

Alternatively, you can add the guard check inside of `manageNodeScanLoop`. Up to you.

Reply:
Thank you, I completely overlooked the call to `manageNodeScanLoop`.
I have moved the additional checking into the function, taking care to return the channel either case, so it cannot break the `go func` that multiplexes producers.

Reply:
The way this is now we are leaking the go routine. The channel is created, but there are no producers attached to it. Also, we are blocking an entire go routine (multiplexer below) waiting for a message that will never arrive.